src/PersistentOrderedMap.mo

```diff
@@ -754,6 +798,36 @@ module {
       }
     };
 
+    public func maxEntry<K, V>(m: Map<K, V>): ?(K, V) {
+      func rightmost(m: Map<K, V>): ?(K, V) {
```

Comment:
We can return `(K, V)` here and create `?rightmost(m)` on the outer call. I'm not sure about performance implications, but in haskell I would say that such code would be more optimisations friendly.

Reply:
fixed